backends/backends.c

```diff
@@ -833,12 +897,53 @@ void *backends_main(void *ptr) {
                 chart_sent_metrics = chart_buffered_metrics;
 
             buffer_flush(b);
-        }
-        else {
-#else
-        {
+        } else
 #endif /* HAVE_KINESIS */
 
+#if HAVE_MONGOC
+        if(do_mongodb) {
+            size_t buffer_len = buffer_strlen(b);
+            size_t sent = 0;
+
+            while(sent < buffer_len) {
+                const char *first_char = buffer_tostring(b);
+
+                debug(D_BACKEND, "BACKEND: mongodb_insert(): uri = %s, database = %s, collection = %s, \
+                      buffer = %zu", mongodb_uri, mongodb_database, mongodb_collection, buffer_len);
+
+                if(!mongodb_insert(first_char)) {
+                    sent += buffer_len;
+                    chart_transmission_successes++;
+                    chart_receptions++;
+                }
```

Comment:
I personally like the else bellow '}', but this is a different pattern from the rest of the file.

Reply:
There are a lot of places in the file where else is below '}'.